rust/nix-expr/src/eval_state.rs

```diff
@@ -50,17 +50,30 @@ pub struct EvalState {
     context: Context,
 }
 impl EvalState {
-    pub fn new(store: Store) -> Result<Self> {
+    pub fn new<'a>(lookup_path: impl Iterator<Item = &'a str>, store: Store) -> Result<Self> {
```

Comment:
Could you put the lookup path second?
I'd consider it to be more optional, so I'd expected it towards the end of the argument list.